src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +40,110 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
+                    auto end  = m_end_mask_set.count(i) ? shapeIn[i] : endPtr[i];
+                    m_outputShape[new_idx] = ov::op::slice::get_sliced_value(shapeIn[i], begin, end, stridePtr[i]);
+                }
+                new_idx += 1;
+            }
+        }
+        return {{m_outputShape}, ShapeInferStatus::success};
+    }
+
+    port_mask_t get_port_mask() const override {
+        return m_port_mask;
+    }
+
+private:
+    const IShapeInfer::port_mask_t m_port_mask;
+    VectorDims m_outputShape;
+    // set
+    const std::unordered_set<int64_t> m_begin_mask_set;
+    const std::unordered_set<int64_t> m_end_mask_set;
+    const std::unordered_set<int64_t> m_new_axis_mask_set;
+    const std::unordered_set<int64_t> m_shrink_axis_mask_set;
+};
+
+class StridedSliceShapeInferFactory : public ShapeInferFactory {
+public:
+    StridedSliceShapeInferFactory(const std::shared_ptr<ov::Node>& op, IShapeInfer::port_mask_t port_mask)
+    : m_op(op), m_port_mask(port_mask) {}
+    ShapeInferPtr makeShapeInfer() const override {
+        if (const auto Slice_op = ov::as_type_ptr<const ov::op::v8::Slice>(m_op)) {
+            return std::make_shared<NgraphShapeInfer>(make_shape_inference(m_op), m_port_mask);
+        } else if (const auto StridedSlice_op = ov::as_type_ptr<const ov::op::v1::StridedSlice>(m_op)) {
+            const auto& ellipsis_mask = StridedSlice_op->get_ellipsis_mask();
+            if (std::count(ellipsis_mask.begin(), ellipsis_mask.end(), 1)) {
+                return std::make_shared<NgraphShapeInfer>(make_shape_inference(m_op), m_port_mask);
+            } else {
+                auto vec_to_set = [](const std::vector<int64_t>& vec){
+                    std::unordered_set<int64_t> to_set;
+                    for (auto i = 0; i < vec.size(); ++i) {
+                        if (vec[i] == 1) {
+                            to_set.emplace(i);
+                        }
+                    }
+                    return to_set;
+                };
+                return std::make_shared<StridedSliceShapeInfer>(
+                    m_port_mask,
+                    m_op->get_output_partial_shape(0).rank().get_length(),
+                    vec_to_set(StridedSlice_op->get_begin_mask()),
+                    vec_to_set(StridedSlice_op->get_end_mask()),
+                    vec_to_set(StridedSlice_op->get_new_axis_mask()),
+                    vec_to_set(StridedSlice_op->get_shrink_axis_mask()));
+            }
+        } else {
+            IE_THROW(NotImplemented) << "not Slice or StridedSlice";
+        }
+    }
+
+private:
+    const std::shared_ptr<ov::Node> m_op;
+    const IShapeInfer::port_mask_t m_port_mask;
+};
+
 StridedSlice::StridedSlice(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr context) :
-        Node(op, context, NgraphShapeInferFactory(op, PortMask(1, 2, 3, 4))) {
+        Node(op, context, StridedSliceShapeInferFactory(op, PortMask(1, 2, 3, 4))) {
```

Comment:
Why we have to pass the port mask? It should be a static field of `StridedSliceShapeInfer`, I mean static constexpr. So it can be passed to the `NgraphShapeInfer` constructor and returned from the `get_port_mask` call.

Reply:
`StridedSliceShapeInferFactory` do not need to have those parameter in the parameter list at all. I propose to do the following:
1. Introduce a static field in the `StridedSliceShapeInfer` as `static constexpr port_mask = PortMask(1, 2, 3, 4);`
2. Reuse this field in the `get_port_mask()` implementation
3. Pass this value to the `NgraphShapeInfer` inside the factory.
This simple action will reduce the size of the shape inference class and the shape infer factory size by `sizeof(port_mask_t)` and remove the unnecessary complexity from the factory.

Reply:
> `StridedSliceShapeInferFactory` do not need to have those parameter in the parameter list at all. I propose to do the following:
> 
> 1. Introduce a static field in the `StridedSliceShapeInfer` as `static constexpr port_mask = PortMask(1, 2, 3, 4);`
> 2. Reuse this field in the `get_port_mask()` implementation
> 3. Pass this value to the `NgraphShapeInfer` inside the factory.
>    This simple action will reduce the size of the shape inference class and the shape infer factory size by `sizeof(port_mask_t)` and remove the unnecessary complexity from the factory.

modified these port_mask in func and class.